src/lib/reactComponentUtils.ts

```diff
@@ -0,0 +1,21 @@
+export const getComponentName = (component: React.ReactType) => {
+  switch (typeof component) {
```

Comment:
N.A. anymore since the implementation was drastically simplified following decision:
https://github.com/stardust-ui/react/pull/496#discussion_r238708722